presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/MetastoreUtil.java

```diff
@@ -896,4 +898,16 @@ private static OptionalLong parse(@Nullable String parameterValue)
 
         return result.build();
     }
+
+    public static Map<String, String> addStatsGeneratedConf(Map<String, String> parameters)
+    {
+        if (parameters.isEmpty()) {
+            return parameters;
+        }
+        ImmutableMap.Builder<String, String> result = ImmutableMap.builder();
+        parameters.entrySet().stream().forEach(entry -> result.put(entry.getKey(),
+                entry.getValue()));
+        result.put(STATS_GENERATED, "true");
+        return result.build();
+    }
```

Comment:
I am trying to understand better. why do we need a separate function ? Can it be integrated in to the `updateStatisticsParameters()` ? Also what is `STATS_GENERATED` telling us ?

Reply:
I'm sorry for the redundant change because there is a different hive version between open source and my environment.
STATS_GENERATED is the parameter to tell the hive to replace all properties of partition with external properties because some properties can't be covered (e.g createTime) when STATS_GENERATED is false .
But STATS_GENERATED is set true in hive-2.1 when alterPartition/alterTable happens, so I delete the separate function now.